instrumentation/github.com/bradfitz/gomemcache/gomemcache.go

```diff
@@ -0,0 +1,226 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gomemcache
+
+import (
+	"context"
+
+	"github.com/bradfitz/gomemcache/memcache"
+
+	otelglobal "go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/standard"
+	oteltrace "go.opentelemetry.io/otel/api/trace"
+)
+
+const (
+	defaultTracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/bradfitz/gomemcache"
+	tracerVersion     = "semver:0.1.0"
+
+	defaultServiceName = "memcached"
+)
+
+// Client is a wrapper around *memcache.Client.
+type Client struct {
+	*memcache.Client
+	cfg *config
+	ctx context.Context
+}
+
+// NewClientWithTracing wraps the provided memcache client to allow
+// tracing of all client operations. Accepts options to set tracer and
+// service name, otherwise uses default values.
+//
+// Every client operation starts a span with appropriate attributes,
+// executes the operation and ends the span (additionally also sets a status
+// error code and message, if an error occurs). Optionally, client context can
+// be set before an operation with the WithContext method.
+func NewClientWithTracing(client *memcache.Client, opts ...Option) *Client {
+	cfg := &config{}
+	for _, o := range opts {
+		o(cfg)
+	}
+
+	if cfg.serviceName == "" {
+		cfg.serviceName = defaultServiceName
+	}
+
+	if cfg.tracer == nil {
+		cfg.tracer = otelglobal.TraceProvider().Tracer(
+			defaultTracerName,
+			oteltrace.WithInstrumentationVersion(tracerVersion),
+		)
+	}
```

Comment:
Can this be changed to accept a `trace.Provider` in the configuration rather than a `trace.Tracer`?  See #196 for details on a change I expect we may wish to expand to all instrumentations.  I think it does make more sense to allow the consumer to specify a provider while the instrumentation remains responsible for configuring the tracer with its name and version.

Reply:
(Same comment.)

Reply:
I see, I adjusted the client configuration to accept `trace.Provider` instead.